### ISSUE TITLE
Mounting on FastAPI deployed on Posit Connect redirects to 127.0.0.1

### ISSUE THREAD

**Issue by tonisama** [Chainlit/chainlit#1604]
**Describe the bug**
I've mounted my Chainlit app as a FastAPI sub-application as outlined here https://docs.chainlit.io/integrations/fastapi.  I've tested this locally and it works as expected. 

I tried to deploy the app as a FastAPI on Posit Connect, and when navigating to /chainlit, I get redirected to 127.0.0.1.  The /docs and /app do work as expected which means FastAPI is correctly set up.

Tried adding environmental variables with no effect:
CHAINLIT_HEADLESS = true
CHAINLIT_NO_ONBOARDING = true
CHAINLIT_PUBLIC_PATH = '/chainlit' 

**To Reproduce**

1. Go to 'https://docs.chainlit.io/integrations/fastapi'
2. Copy all code into the appropriate files in a directory
3. Deploy the app using the command line: 'rsconnect deploy fastapi --entrypoint main:app --server [your_server_URL] --api-key [your_API_key] ./
4. Attempt to navigate to your deployed app at: '[your_app_url]/chainlit'

**Expected behavior**
Expect to see the app load.  Instead, redirected to 127.0.0.1/[rest_of_URL]/chainlit and a 'page not found' error.

Note if mounting locally using uvicorn, the app loads fine.

**Comment by willydouhard** [comment]
When started locally on the sub path is it working? What version of Chainlit are you using?

**Comment by tonisama** [comment]
> When started locally on the sub path is it working? What version of Chainlit are you using?

Yes, it works locally, either by using "chainlit run" or by running a FastAPI with uvicorn web server and navigating to "localhost/chainlit"  I'm using 1.3.2.

**Comment by tonisama** [comment]
Posit Support replied back saying "Chainlit doesn't appear to correctly handle running behind a proxy with a non-empty root path."  I'm not sure I necessarily buy this.

**Comment by fvanrossum** [comment]
I wanted to follow-up on this as I also spent a while running into this issue.

The following fixed the issue. So the problem seems to be arising from the fact that the server or chainlit does not properly use the root path.

app.py
```

import os
from fastapi import FastAPI, Request
from chainlit.utils import mount_chainlit

import os 

# print out all environment variables
for key, value in os.environ.items():
    print(f"{key}: {value}")

# Get the root path from the environment (Posit Connect sets this)
server_url = os.environ.get("CONNECT_SERVER")
guid = os.environ.get("CONNECT_CONTENT_GUID")
root_path = f"/content/{guid}" if guid else ""

print(f"Root path: {root_path}")

app = FastAPI(root_path=root_path)

@app.get("/debug-headers")
async def debug_headers(request: Request):
    return dict(request.headers)

@app.get("/app")
def read_main():
    return {"message": "Hello World from main app"}


mount_chainlit(app=app, target="mycl_app.py", path=f"/")


print("chainlit root path", os.getenv("CHAINLIT_ROOT_PATH"))
print("Chainlit parent root path", os.getenv("CHAINLIT_PARENT_ROOT_PATH"))
print(app.root_path)
print(app.root_path_in_servers)
```

mycl_app.py
```
import chainlit as cl

@cl.on_chat_start
async def main():
    await cl.Message(content="Hello World").send()
```
